tmail-backend/tmail-third-party/openpaas/src/main/java/com/linagora/tmail/carddav/CardDavClient.java

```diff
@@ -0,0 +1,92 @@
+package com.linagora.tmail.carddav;
+
+import java.time.Duration;
+import java.util.function.Function;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.auth.UsernamePasswordCredentials;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Preconditions;
+import com.linagora.tmail.HttpUtils;
+import com.linagora.tmail.configuration.CardDavConfiguration;
+
+import io.netty.buffer.Unpooled;
+import io.netty.handler.codec.http.HttpHeaderNames;
+import io.netty.handler.ssl.SslContextBuilder;
+import io.netty.handler.ssl.util.InsecureTrustManagerFactory;
+import reactor.core.publisher.Mono;
+import reactor.netty.http.client.HttpClient;
+
+public interface CardDavClient {
+
+    Mono<Boolean> existsCollectedContact(String username, String userId, String collectedId);
+
+    Mono<Void> createCollectedContact(String username, String userId, CardDavCreationObjectRequest creationObjectRequest);
+
+    class OpenpaasCardDavClient implements CardDavClient {
+
+        private static final Logger LOGGER = LoggerFactory.getLogger(OpenpaasCardDavClient.class);
+        private static final Duration RESPONSE_TIMEOUT_DEFAULT = Duration.ofSeconds(10);
+        private static final String COLLECTED_ADDRESS_BOOK_PATH = "/addressbooks/%s/collected/%s.vcf";
+
+        private final HttpClient client;
+
+        private final Function<String, UsernamePasswordCredentials> adminCredentialWithDelegatedFunction;
+
+        public OpenpaasCardDavClient(CardDavConfiguration cardDavConfiguration) {
+            this.client = HttpClient.create()
+                .baseUrl(cardDavConfiguration.baseUrl().toString())
+                .headers(headers -> headers.add(HttpHeaderNames.ACCEPT, "application/vcard+json"))
+                .responseTimeout(cardDavConfiguration.responseTimeout().orElse(RESPONSE_TIMEOUT_DEFAULT));
+
+            cardDavConfiguration.trustAllSslCerts().ifPresent(trustAllSslCerts ->
+                client.secure(sslContextSpec -> sslContextSpec.sslContext(
```

Comment:
Method `secure` would create a duplicate and apply SSL config on that instance, so this code does not have any effect, unless I'm missing something?

Cc @vttranlina